CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/panes/DataAccessSearchProviderNGTest.java

```diff
@@ -0,0 +1,324 @@
+        // Mock of DataAccessPane will return a blank new tab when called
+        try (MockedStatic<DataAccessUtilities> mockedStatic2 = Mockito.mockStatic(DataAccessUtilities.class)) {
+            when(daPane.getCurrentTab()).thenReturn(new Tab());
+            mockedStatic2.when(() -> DataAccessUtilities.getDataAccessPane()).thenReturn(daPane);
+        }
+
+        // Mock request to return the selected text
+        when(request.getText()).thenReturn("Select");
+
+        // Return a valid response when results are added that match the expected
+        when(response.addResult(Mockito.anyObject(), Mockito.eq("Select Top N"))).thenReturn(true);
+        when(response.addResult(Mockito.anyObject(), Mockito.eq("Select All"))).thenReturn(true);
+
+        DataAccessSearchProvider instance = new DataAccessSearchProvider();
+        instance.evaluate(request, response);
+
+        // Verify that addResult was called on the correct plugins
+        verify(response, times(1)).addResult(Mockito.anyObject(), Mockito.eq("Select Top N"));
+        verify(response, times(1)).addResult(Mockito.anyObject(), Mockito.eq("Select All"));
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    ///////////////////  Start of PluginDisplayer Tests  ///////////////////////
+    ////////////////////////////////////////////////////////////////////////////
+    /**
+     * Test of run method, of class PluginDisplayer.
+     */
+    @Test
+    public void testRun1() {
+        System.out.println("testRun1 testing query phase pane was successfully returned");
+
+        // Setting up mocks
+        MockitoAnnotations.initMocks(this);
+        daPane = mock(DataAccessPane.class);
+        qpPane = mock(QueryPhasePane.class);
+
+        // Do nothing when the plugin is called to expand
+        // May never get here with the code existing in the Platform thread.
+        doNothing().when(qpPane).expandPlugin(Mockito.eq("SelectTopN"));
+
+        // Mock the static method getQueryPhasePane to return the mocked QueryPhasePane
+        try (MockedStatic<DataAccessPane> mockedStatic = Mockito.mockStatic(DataAccessPane.class)) {
+            mockedStatic.when(() -> DataAccessPane.getQueryPhasePane(Mockito.anyObject())).thenReturn(qpPane);
+
+            // Return a new tab when the DataAccessPane mock is prompted for getCurrentTab
+            when(daPane.getCurrentTab()).thenReturn(new Tab());
+
+            // Mock the static method getDataAccessPane and return the mocked DataAccessPane
+            try (MockedStatic<DataAccessUtilities> mockedStatic2 = Mockito.mockStatic(DataAccessUtilities.class)) {
+                mockedStatic2.when(() -> DataAccessUtilities.getDataAccessPane()).thenReturn(daPane);
+
+                try {
+                    PluginDisplayer pd = new PluginDisplayer("SelectTopN");
+                    pd.run();
+                } catch (final IllegalStateException ex) {
```

Comment:
@antares1470 I've just caught the exception for the toolkit not instantiated and allowed the code to continue in this situation as I'm only able to test the verify() calls as opposed to a return value.
I was thinking about surrounding it in the if headless block but that would not test anything in the method that way.

Reply:
Perhaps this is a better solution than checking for headless. :+1: